pkg/controller/shoot.go

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+)
+
+// Shoot provides convenience methods for getting various Shoot properties.
+type Shoot struct {
+	*gardenv1beta1.Shoot
+}
+
+// GetKubernetesVersion returns this shoot's Kubernetes version.
+func (s *Shoot) GetKubernetesVersion() string {
+	return s.Shoot.Spec.Kubernetes.Version
+}
+
+// GetPodNetwork returns this shoot's pod network CIDR.
+func (s *Shoot) GetPodNetwork() gardencorev1alpha1.CIDR {
+	cloud := s.Shoot.Spec.Cloud
+	if cloud.AWS != nil {
```

Comment:
I'd recommend a switch here.

Reply:
Switch on what exactly? I thought about a switch and looked at how similar stuff is done in Gardener, I was only able to find `nil` checks. It's also more reliable in my opinion. But I can of course write it as a switch if you point me to the right example.

Reply:
If you have lots of `if` / `else` blocks, you can just switch on 'nothing' and write your `if`s as `case`s:
```go
switch {
  case a:
  case b:
}
```

Reply:
Yes, you can do this, but it's by no means a widely used style. It's not present in any of the common Go coding style guidelines I know of. Do you have a coding guideline for Gardener that actually encourages such a style? If not, I would recommend to write one as you can't expect  contributors to know that they should write their code like this.

Reply:
Reworked as asked.